logstash-core/src/main/java/org/logstash/ext/JRubyWrappedWriteClientExt.java

```diff
@@ -164,4 +181,76 @@ public void push(Map<String, Object> event) {
         writeClient.push(event);
         incrementTimers(start);
     }
+
+    private void registerFlowMetrics(final ThreadContext context,
```

Comment:
It is unclear to me why we are doing this in the queue write client _in addition to_ in the pipeline implementation that already exists on the branch. The flow metrics specified for this effort are pipeline-specific, powered by metrics that we already register in the pipeline. Do we have an alternate pipeline implementation for tests that is failing to register them and/or doesn't inherit from our abstract base pipeline?

I'll dig in more in the morning, but if you have more context it would be helpful.

Reply:
Ah, please ignore this change. Out of curiosity tried to implement and see the plugin level flow metrics. Yesterday I was confused with `JRubyAbstractQueueWriteClientExt` class and figured out that it is a [client for plugin workers](https://github.com/elastic/logstash/blob/04b8ab240e0e26bf87d7d21da3fe82a56d33cc73/logstash-core/lib/logstash/java_pipeline.rb#L411). I simply copied and paste to see if our flow metric captures input plugin flow. :) no luck, we will deep dive in our next phases.

Reply:
Ah. The "input workers" (which we generally don't refer to as "workers" without qualification). Each input plugin gets its own "input worker", with its own queue write client that it uses for writing events to the queue and recording metrics about interactions with the queue (including metrics for that individual plugin).